src/server/pfs/server/master.go

```diff
@@ -95,8 +123,18 @@ func (d *driver) finishCommits(ctx context.Context) error {
 		ctx, cancel := context.WithCancel(ctx)
 		repos[key] = cancel
 		go func() {
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
 			backoff.RetryUntilCancel(ctx, func() error { //nolint:errcheck
-				return d.finishRepoCommits(ctx, compactor, key)
+				lockCtx, err := ring.Lock(ctx, lockPrefix)
+				if err != nil {
+					return errors.Wrap(err, "error locking repo lock")
+				}
+				defer func() {
+					err := ring.Unlock(lockPrefix)
+					retErr = multierror.Append(retErr, errors.Wrap(err, "error unlocking"))
```

Comment:
I don't think this is going to be correct for the nil case. I think it should be something like this:
```
if err := ring.Unlock(lockPrefix); err != nil {
  if retErr == nil {
    retErr = err
  } else {
    retErr = multierror.Append(retErr, errors.Wrap(err, "error unlocking"))
  }
}
```